compositor/sessionmanager/sessionmanager.cpp

```diff
@@ -320,9 +285,6 @@ void SessionManager::requestHybridSleep()
     Q_EMIT hybridSleepRequested();
 }
 
-void SessionManager::cancelShutdownRequest()
-{
-    Q_EMIT shutdownRequestCanceled();
-}
+void SessionManager::cancelShutdownRequest() { Q_EMIT shutdownRequestCanceled(); }
 
-#include "moc_sessionmanager.cpp"
+#include "sessionmanager.moc"
```

Comment:
Right, but if Q_OBJECT was used in a C++ file, it's going to be named `*.moc` per the link you shared. If it's not included in a source file, it'll be compiled automatically, so we don't normally need to add it at the bottom of our source files.

However... when Q_OBJECT is used in a C++ file like I'm doing here, without the include or when including `moc_*.cpp`,  I get:

```
[  9%] Automatic moc and rcc for target hawaii
AUTOGEN: error: /home/mspencer/Developer/qmlos/shell/compositor/sessionmanager/sessionmanager.cpp: The file contains a Q_OBJECT macro, but does not include "sessionmanager.moc" !

make[2]: *** [compositor/CMakeFiles/hawaii_automoc.dir/build.make:58: compositor/CMakeFiles/hawaii_automoc] Error 1
make[1]: *** [CMakeFiles/Makefile2:329: compositor/CMakeFiles/hawaii_automoc.dir/all] Error 2
```

So... we don't normally need to include the moc files ourselves, unless `Q_OBJECT` is used in a C++ file, in which case moc requires we include the file named `*.moc`.